scripts/irods/upgrade_configuration.py

```diff
@@ -33,7 +33,89 @@ def requires_upgrade(irods_config):
     new_version_tuple = lib.version_string_to_tuple(new_version['irods_version'])
     old_version_tuple = lib.version_string_to_tuple(irods_config.version['irods_version'])
 
-    return old_version_tuple != new_version_tuple
+    return old_version_tuple < new_version_tuple
+
+def raise_exception_if_downgrade_is_detected(irods_config):
+    l = logging.getLogger(__name__)
+
+    # On a new installation, the only file that will exist is version.json.dist.
+    # version.json.previous only exists on upgrade.
+
+    # Look for a ".previous" file. Start with the latest form of the file and work
+    # backwards based on iRODS versions.
+    version_file_path_previous = None
+    version_file_paths = [
+        paths.version_path() + '.previous',                             # >= 4.3.x
+        os.path.join(paths.irods_directory(), 'VERSION.json.previous'), # >= 4.1.x
+        os.path.join(paths.irods_directory(), 'VERSION.previous')       # <  4.1.x
+    ]
+    for path in version_file_paths:
+        l.debug('Checking if [%s] exists.', path)
+        if os.path.exists(path):
+            version_file_path_previous = path
+            break
+
+    if version_file_path_previous is None:
+        # This is a brand new installation of iRODS. No version of iRODS existed on the
+        # computer before now. No risk of downgrading the server.
+        l.info('This is a new installation of iRODS. No downgrade issue present.')
+        return
+
+    l.debug('Found [%s].', version_file_path_previous)
+
+    # TODO Remove this block since it doesn't make since for iRODS 5.
+    # There will never be a case where old ".dist" files exist during an upgrade.
```

Comment:
dist files are always there because we 'distribute' them?   so not a matter of 'old' or 'upgrade'?

Reply:
Removed this code.